docs/designs/inventory-design.md

```diff
@@ -0,0 +1,180 @@
+
+Adding/editing makes another API call to Supabase to create a new row in the SQL table or by updating existing row. The application then makes another GET request to update the data state and display the new array.
+
+## 6. Security and Compliance
+
+### 6.1 Security Considerations
+Authorization is based on the user's account level, which is determined with their Azure OAuth. Access to read/edit Supabase requires users to have an anon-key provided by Supabase and pre-approved by admin level users.
+
+Currently, the anon-key is input on the .env-local file, which is served to the client upon access. This may be changed later via server-side encryption to prevent exposure.
+
+### 6.2 Compliance Requirements
+- Ensure compliance with GDPR for handling volunteer data.
+  - **Data Minimization**: Only collect data that is necessary for the operation of the system. For example, storing name, email, and PIN is necessary for authentication.
+
+## 7. Performance Considerations
+
+### 7.1 Performance Requirements
+Fetching data and populating the inventory page upon initialization should be quick. Filter responses are set to 300ms so as to not continuously make filter requests when typing in the search input. Adding/updating the inventory page is optimized for fast response time, with 'status' column being updated on the database side.
+
+
+### 7.2 Scalability
+- The architecture should support adding more items and scaling up the database as needed.
+- The system should handle increased traffic during peak volunteer sign-in times.
+
+## 8. Deployment Considerations
+
+### 8.1 Deployment Strategy
+- Deploy the web application on a cloud platform (e.g., Azure Web Services).
+- Integrate Azure OAuth configurations for authentication.
+- Ensure database backups and failover strategies are in place.
+
+### 8.2 Maintenance
+- Regularly update and patch the OAuth integration for security.
+- Monitor the database for performance and maintain data integrity.
+
+## 9. Risk Analysis
+Anon-key exposure on the client side could lead to unwanted read/edit access. Mitigation for this would be to reset the anon-key and rollback affected data. Alternatively, would be to create a server side for application so that it can be hidden.
+Identify potential risks in the architecture and provide mitigation strategies.
+
+## 10. Database Schema
+
+### Inventory
+| Column Name      | Data Type | Description                          |
+|------------------|-----------|--------------------------------------|
+| `id`             | INT       | Primary key, auto-increment.         |
+| `item`     | VARCHAR   | Item's name.              |
+| `type`  | VARCHAR   | Donation Type (donation or welcome basket)  |
```

Comment:
is this always either donation or basket? Should we make it an enum datatype (or add a CHECK constraint to limit freedom):

mycol VARCHAR(10) NOT NULL CHECK (mycol IN('Donation', 'Basket', 'Unknown'))

Reply:
It's always donation or welcome basket. But I left as varchar because maybe they'll add another type.